scipy/io/wavfile.py

```diff
@@ -208,48 +233,49 @@ def read(filename, mmap=False):
         fid = open(filename, 'rb')
 
     try:
-        fsize, is_big_endian = _read_riff_chunk(fid)
+        file_size, is_big_endian = _read_riff_chunk(fid)
         fmt_chunk_received = False
-        noc = 1
-        bits = 8
-        comp = WAVE_FORMAT_PCM
-        while (fid.tell() < fsize):
+        channels = 1
+        bit_depth = 8
+        format_tag = WAVE_FORMAT_PCM
+        while fid.tell() < file_size:
             # read the next chunk
             chunk_id = fid.read(4)
             if chunk_id == b'fmt ':
                 fmt_chunk_received = True
-                size, comp, noc, rate, sbytes, ba, bits = _read_fmt_chunk(
-                                        fid, is_big_endian=is_big_endian)
-                if bits not in (8, 16, 32, 64, 128):
+                fmt_chunk = _read_fmt_chunk(fid, is_big_endian)
+                format_tag, channels, fs = fmt_chunk[1:4]
+                bit_depth = fmt_chunk[6]
+                if bit_depth not in (8, 16, 32, 64, 128):
                     raise ValueError("Unsupported bit depth: the wav file "
-                                     "has {}-bit data.".format(bits))
+                                     "has {}-bit data.".format(bit_depth))
             elif chunk_id == b'fact':
-                _skip_unknown_chunk(fid, is_big_endian=is_big_endian)
+                _skip_unknown_chunk(fid, is_big_endian)
             elif chunk_id == b'data':
                 if not fmt_chunk_received:
                     raise ValueError("No fmt chunk before data")
-                data = _read_data_chunk(fid, comp, noc, bits,
-                                        is_big_endian=is_big_endian, mmap=mmap)
+                data = _read_data_chunk(fid, format_tag, channels, bit_depth,
+                                        is_big_endian, mmap)
             elif chunk_id == b'LIST':
                 # Someday this could be handled properly but for now skip it
-                _skip_unknown_chunk(fid, is_big_endian=is_big_endian)
+                _skip_unknown_chunk(fid, is_big_endian)
             elif chunk_id in (b'JUNK', b'Fake'):
                 # Skip alignment chunks without warning
-                _skip_unknown_chunk(fid, is_big_endian=is_big_endian)
+                _skip_unknown_chunk(fid, is_big_endian)
             else:
                 warnings.warn("Chunk (non-data) not understood, skipping it.",
                               WavFileWarning)
-                _skip_unknown_chunk(fid, is_big_endian=is_big_endian)
+                _skip_unknown_chunk(fid, is_big_endian)
     finally:
         if not hasattr(filename, 'read'):
             fid.close()
         else:
             fid.seek(0)
 
-    return rate, data
+    return fs, data
 
 
-def write(filename, rate, data):
+def write(filename, fs, data):
```

Comment:
As this function is technically public, I don't think we can change this without deprecation. You could do `write(filename, fs, data, rate=None)` if you want, and than arg-based and kwarg-based as `write(filename, rate=X)` would still work